tasks/merge/README.md

```diff
@@ -0,0 +1,7 @@
+# Merge task
+
+This task requires merging many small expression matrices into one large one.
```

Comment:
Do we assume all expression matrices have the same number of rows (genes) and also the index of genes are exactly the same (so there is no need to shuffle rows of each matrix before  concatenating them?
If so, let's state these assumption to make the task more concrete.

Reply:
a small aside is that distributed processing of expression matrices is almost definitely going to treat rows as cells and columns as genes; iirc scanpy/anndata do this already, while seurat/loom do gene-rows. The world's extant distributed-processing machinery distinguishes between row- and column-semantics in a way that fairly strongly motivates this, though happy to get into more details about it if that seems bad/controversial!

Reply:
@mckinsel I think @ryan-williams is expanding a bit more, but here is the overarching way I think about it: most distributed computing systems are about appending (rows) to data sets and having immutable data. So people generally arrange it so that the dimension that will grow rapidly (number of cells) is the rows while the dimension that's relatively constant is the columns.

Reply:
@mckinsel good point, this does come up a lot and is worth communicating more about; [here's an attempt at a longer analysis](https://gist.github.com/ryan-williams/e942472ef581b36ec76548ebd84ee953), lmk what you think (here, on the gist, or e.g. in slack wfm)